extensions/hp-extension/src/custom-viewport/setViewportZoomPan.ts

```diff
@@ -5,30 +5,67 @@
  * @param value 
  * @param viewport 
  */
-const setViewportZoomPan = (id,value,viewport) => {
-  const {zoom,pan, flipHorizontal, flipVertical} = value;
-  if( zoom && zoom!==1 ) viewport.setZoom(zoom);
-  if( pan ) viewport.setPan(pan);
-  if( flipHorizontal || flipVertical ) {
+const setViewportZoomPan = (id, value, viewport) => {
+  const { zoom, pan, flipHorizontal, flipVertical } = value;
+  const { imageCanvasPoint, initialDisplayArea } = value;
+
+  const camera = viewport.getCamera();
+  console.log("Camera=", JSON.stringify(camera));
+
+  // Set the base values - assume height fits for now
+  let initialZoom = 1;
+  if (initialDisplayArea) {
+    // TODO - use the correct calculation to actually fit things properly
+    // such that both initialDisplayArea's fit
+    initialZoom = 1 / Math.min(initialDisplayArea[0], initialDisplayArea[1]);
+    viewport.setZoom(initialZoom, true);
+  }
+
+  const imageData = viewport.getImageData();
+
+  if (imageCanvasPoint && imageData) {
+    const canvas = viewport.getCanvas();
+    const canvasX = imageCanvasPoint[2] ?? 0.5;
+    const canvasY = imageCanvasPoint[3] ?? 0.5;
+    const canvasPanX = canvas.width * (canvasX - 0.5);
+    const canvasPanY = canvas.height * (canvasY - 0.5);
+   
+    const canvasZero = viewport.worldToCanvas([0,0,0]);
+    const canvasEdge = viewport.worldToCanvas(imageData.dimensions);
+    const canvasImage = [canvasEdge[0]-canvasZero[0], canvasEdge[1]-canvasZero[1]];
```

Comment:
there is no need on creating an array to later do a destructuring.

WE can just define imageWidth directly, no? const imageWidth = canvasEdge[0]-canvasZero[0];